packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -293,6 +288,8 @@ export function createVM<HostNode, HostElement>(
         cmpTemplate: null,
 
         renderMode: def.renderMode,
+        shadowMode: computeShadowMode(def, owner),
+        nearestSRMode: owner?.shadowRoot ? owner.shadowMode : owner?.nearestSRMode || null,
```

Comment:
1. `nearestSRMode` will be null for root elements correct? Is there a simpler way to find that?
2. This is not considering light dom elements. Previously for light dom elements, the algo traversed up to an ancestor whose renderMode is not Light.

Reply:
> This is not considering light dom elements. Previously for light dom elements, the algo traversed up to an ancestor whose renderMode is not Light.

I think for that case, the child light component's `nearestSRMode` is supposed to be the `nearestSRMode` of the parent. So it should be passed all the way down from the shadow container through any number of light descendants.

Reply:
```suggestion
        nearestSRMode: owner?.shadowRoot ? owner.shadowMode : (owner?.nearestSRMode ?? null),
```

I'm surprised it didn't break any tests, but using `||` is also unsafe here since the srMode may be `0`. I'd also suggest adding the parens to make the logic a bit easier to read.